test/e2e/hsts_policy_test.go

```diff
@@ -0,0 +1,171 @@
+		// Remove the HSTS policies from the ingress config
+		ing.Spec.RequiredHSTSPolicies = nil
+		if err := kclient.Update(context.TODO(), ing); err != nil {
+			t.Fatalf("failed to restore ingress config: %v", err)
+		}
+	}()
+
+	p := ing.Spec.RequiredHSTSPolicies[0]
+	t.Logf("created a RequiredHSTSPolicy with DomainPatterns: %v,\n preload policy: %s,\n includeSubDomains policy: %s,\n largest age: %d,\n smallest age: %d\n", p.DomainPatterns, p.PreloadPolicy, p.IncludeSubDomainsPolicy, *p.MaxAge.LargestMaxAge, *p.MaxAge.SmallestMaxAge)
+
+	// Use the same namespace for route, service, and pod
+	ns := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "hsts-policy-namespace",
+		},
+	}
+	if err := kclient.Create(context.TODO(), ns); err != nil {
+		t.Fatalf("failed to create namespace: %v", err)
+	}
+	defer func() {
+		// this will cleanup all components in this namespace
+		if err := kclient.Delete(context.TODO(), ns); err != nil {
+			t.Fatalf("failed to delete test namespace %s: %v", ns.Name, err)
+		}
+	}()
+
+	// Create pod
+	echoPod := buildEchoPod("hsts-policy-echo", ns.Name)
+	if err := kclient.Create(context.TODO(), echoPod); err != nil {
+		t.Fatalf("failed to create pod %s/%s: %v", ns, echoPod.Name, err)
+	}
+
+	// Create service
+	echoService := buildEchoService(echoPod.Name, ns.Name, echoPod.ObjectMeta.Labels)
+	if err := kclient.Create(context.TODO(), echoService); err != nil {
+		t.Fatalf("failed to create service %s/%s: %v", echoService.Namespace, echoService.Name, err)
+	}
+
+	// Create a route that should fail the HSTS policy validation
+	t.Logf("creating an invalid route at %s", time.Now().Format(time.StampMilli))
+	invalidRoute := buildRouteWithHSTS(echoPod.Name+"2", echoPod.Namespace, echoService.Name, icName.Name+"."+appsDomain, "max-age=99999999")
+	if err := kclient.Create(context.TODO(), invalidRoute); err == nil {
+		t.Fatalf("failed to reject an invalid route %s/%s, max-age 99999999", invalidRoute.Namespace, invalidRoute.Name)
+	} else {
+		t.Logf("rejected an invalid route at %s: %s/%s with annotation %s: %v", time.Now().Format(time.StampMilli), invalidRoute.Namespace, invalidRoute.Name, invalidRoute.Annotations, err)
+	}
+
+	// Create a route that should pass the HSTS policy validation
+	exampleHeader := "max-age=0;preload;includesubdomains"
+	t.Logf("creating a valid route at %s", time.Now().Format(time.StampMilli))
+	validRoute := buildRouteWithHSTS(echoPod.Name, echoPod.Namespace, echoService.Name, icName.Name+"2."+appsDomain, exampleHeader)
+	if err := kclient.Create(context.TODO(), validRoute); err != nil {
+		t.Fatalf("failed to create a valid route %s/%s: %v", validRoute.Namespace, validRoute.Name, err)
+	} else {
+		t.Logf("created a valid route at %s: %s/%s with annotation %s", time.Now().Format(time.StampMilli), validRoute.Namespace, validRoute.Name, validRoute.Annotations)
+	}
+
+	// Wait for route host to resolve
+	if err := wait.PollImmediate(10*time.Second, 5*time.Minute, func() (bool, error) {
```

Comment:
Do we still need to do this? We've using the default ingress controller which will have DNS already published. And below when we go on to assert with the `http.Get()` we also wait 5 minutes. This seems unnecessary now.

Reply:
It fixed an issue before, was not aware that issue was now moot.  Will try a test.

Reply:
done